src/Rfc6455/MessageProcessor.php

```diff
@@ -83,6 +82,15 @@ public function onData(string $data, ConnectionInterface $socket, Message $messa
                     try {
                         $frame = new Frame($message->getBuffer());
 
+                        if ($frame->getOpcode() === Frame::OP_CONTINUE && $message->countFrames() === 0) {
+                            throw new ProtocolErrorException('The first frame cannot be a continuation frame');
+                        }
+
+                        if ($frame->getOpcode() === Frame::OP_TEXT && $message->countFrames() >= 1) {
+                            throw new ProtocolErrorException('When there are fragmented frames, the next frame cannot
```

Comment:
`When the Message is fragmented in many frames the only frame that can be a something else than an continue frame is the first`